gslb/ingestion/gslb.go

```diff
@@ -899,18 +899,15 @@ func InitializeMemberCluster(cfg *restclient.Config, cluster KubeClusterDetails,
 	clients[cluster.clusterName] = kubeClient
 
 	var aviCtrl GSLBMemberController
-	if gslbutils.GetCustomFqdnMode() {
-		akoInformerFactory := akoinformer.NewSharedInformerFactory(crdClient, time.Second*30)
-		hostRuleInformer := akoInformerFactory.Ako().V1alpha1().HostRules()
+	akoInformerFactory := akoinformer.NewSharedInformerFactory(crdClient, time.Second*30)
+	hostRuleInformer := akoInformerFactory.Ako().V1alpha1().HostRules()
 
-		aviCtrl = GetGSLBMemberController(cluster.clusterName, informerInstance, &hostRuleInformer)
-		_, err = crdClient.AkoV1alpha1().HostRules("").List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("HostRule API not available for cluster: %v", err)
-		}
-	} else {
-		aviCtrl = GetGSLBMemberController(cluster.clusterName, informerInstance, nil)
+	aviCtrl = GetGSLBMemberController(cluster.clusterName, informerInstance, &hostRuleInformer)
+	_, err = crdClient.AkoV1alpha1().HostRules("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
please check if we limit the listing of hr objects to one since we are checking whether the hostrule API is available or not only.

`metav1.ListOptions{}` has a field called `Limit`, could you please check whether this can help?

Reply:
As discussed, will address this in the future.